### PR TITLE
Update libssh2 for Conan 2.0

### PR DESCRIPTION
**Pull request by dheater**
Specify library name and version:  **libssh2/1.10.0**

Update for Conan v2
Add support for libssh2 version 1.11.0

closes: #17874
closes: #16644

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by dheater** [comment]
Marking as draft because the `mbedtls` option is not working

**Comment by gegles** [comment]
Any update on this? Much appreciated...

**Comment by gegles** [comment]
@dheater, it might be too much to ask, but it would be great if you could also add the latest version (i.e. [1.11.0](https://www.libssh2.org/)) to this PR ... or, if nothing else, make sure your updated recipe proactively works with it.

I just tried to add it to the existing v1 recipe and the fact that conan injects a top-level CMakeLists.txt does screw things up a bit... So I am hoping with your changes it may "just work" as you've removed that CMakeList.txt... I also don't think the current patch applies successfully, but it may also not be needed anymore.. Thx!

If not you, than I can work on subsequent PR to add the latest version once this one is merged. thx!

**Comment by gegles** [comment]
@dheater regarding the latest build errors, you probably need to configure the [BUILD_STATIC_LIBS](https://github.com/libssh2/libssh2/blob/d29eea1d29aa4902a584b81018545be718f6ab55/CMakeLists.txt#L58) and the BUILD_SHARED_LIBS options depending on `shared = True or False` .... in order to only build the right type...

**Comment by dheater** [comment]
@gegles I think I've almost got this one.  The remaining problem is that it is not finding the `libssh2.dll` file on Conan v1 with shared=True.  The error doesn't show in CI, but I get a dialog on Windows when run from cmd shell (not PowerShell).

Sounds like a simple path problem but I do get the `.dll` in the path `/.conan/data/libssh2/<version>/_/_/<hash>/bin/libssh2.dll`

Any ideas?

File: recipes/libssh2/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,7 @@
 cmake_minimum_required(VERSION 3.1)
 project(test_package C)
 
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup(TARGETS)
-
-find_package(Libssh2 REQUIRED CONFIG)
+find_package(libssh2 REQUIRED CONFIG)
```
Comment (dheater) [inline recipes/libssh2/all/test_package/CMakeLists.txt:4]: This change of the `find_package` name from `Libssh2` to `libssh2` is going to break some people. I have tried several ways of keeping `Libssh2` but for some reason my attempts have failed in CI (works locally for me). If someone knows how to fix this, please do so. Better to not break compatibility, even through I think the lowercase version is more consistent and less surprising.

File: recipes/libssh2/all/conandata.yml [hunk 2]
```diff
@@ -14,13 +17,9 @@ sources:
 patches:
   "1.10.0":
     - patch_file: "patches/0001-threads.patch"
```
Comment (gegles) [inline recipes/libssh2/all/conandata.yml:19]: Would you be able to provide a `patch_description` and `patch_type` field so we can get rid of those warnings? Thx!

**Review by gegles — APPROVED** [review]
LGTM, thanks @dheater!

**Review by jcar87 — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**